### ISSUE TITLE
Getting structured ErrorNon2XXStatus fields

### ISSUE THREAD

**Issue by Marcosaurios** [michimani/gotwi#202]
Hi!
First of all thanks for this amazing API! I have been playing around for a while and feels super useful and working really nice!

One question which I stumbled upon, if I follow/unfollow over 50 users, twitter sends a rate limit of 15 min. However, the way is implemented the `ErrorNon2XXStatus`, is only outputting the error in a string format [see here](https://github.com/michimani/gotwi/blob/main/client.go#L157).

Is there any way of accessing the underlying data in `ErrorNon2XXStatus` struct (like for example the `RateLimitInfo` field), different than extracting the info from the string? I see the error is thrown as error struct, which is nice because the functions which calls `c.CallAPI`function can keep throwing the error easily, but I wonder if there was a different way to thrown this specific error in a more detailed way.

 Thanks!

**Comment by michimani** [comment]
@Marcosaurios 

First of all, thank you very much for using this library!
And thank you for your valuable feedback.

As you have pointed out, each field of the `Non2XXError` structure should have been accessible to the users of the library.

I think there are two ways to do this.
The first is to propagate `Non2XXError` from `CallAPI()` so that there are three return values from each API call (e.g. `FollowsFollowingPost()`) like bellow.

```go
func FollowsFollowers(ctx context.Context, c *gotwi.GotwiClient, p *types.FollowsFollowersParams) (*types.FollowsFollowersResponse, *resources.Non2XXError, error) {
	res := &types.FollowsFollowersResponse{}
	not200err, err := c.CallAPI(ctx, FollowsFollowersEndpoint, "GET", p, res)
	if err != nil {
		return nil, nil, err
	}

	if not200err != nil {
		return nil, not200err, nil
	}

	return res, nil, nil
}
```

The second is to keep the two return values and return a structure like the one below.

```go
type Result struct {
	APIResponse util.Response
	Non2XXErr *resources.Non2XXError
}

func FollowsFollowers(ctx context.Context, c *gotwi.GotwiClient, p *types.FollowsFollowersParams) (*Result, error) {
	res := &types.FollowsFollowersResponse{}
	not200err, err := c.CallAPI(ctx, FollowsFollowersEndpoint, "GET", p, res)
	if err != nil {
		return nil, err
	}

	return &Result {
		APIResponse: res,
		Non2XXErr: not200err,
	}
}
```


If you like, please let me know what you think.

Thank you!

**Comment by Marcosaurios** [comment]
Hey @michimani thank you for the quick feedback! 

I see the second option better, it's still an error so making another error returning value I would say is too counterintuitive. 

However I was thinking about the possibility of being able to extend the basic Go Error interface in a new struct called error or whatever for this library, and adding some extra fields there (like if it was a timeout, ratelimit or whatever). I got the idea from best practices in go.dev blog [go-error-handling](https://go.dev/blog/error-handling-and-go) (see the `net` error implementation):

```golang
package net

type Error interface {
    error
    Timeout() bool   // Is the error a timeout?
    Temporary() bool // Is the error temporary?
}
```

So we could do something like: 
```golang
type Error struct {
   error
   Non2XXerr
}
```

Or destructuring the whole Non2XXerr there, I don't know. I thought this way could be nice because we don't need to modify the returning error type for the rest of method's signatures, because the implemented error interface is in that struct, right? 

Pd: you can label this issue as question. 
Thanks for your ideas!

**Comment by michimani** [comment]
@Marcosaurios Thanks for the quick response too! And thanks for the comments on my idea.

The example of extending the Error interface looks good to me.
I'll look into implementing it this way.

Thanks for the useful information!

**Comment by michimani** [comment]
Hey, @Marcosaurios 

I've finished fixing the parts you pointed out and have released it as `v0.10.0`.
You can access more detailed information by casting the error returned by the function of each API call with the extended error type. (Have a look at the updated README).

Thanks to your ideas, we were able to make this product better.  
Thank you so much!

**Comment by Marcosaurios** [comment]
Hey @michimani ,

Really good implementation! This was exactly what I was thinking, thank you so much for the feature!
